cypress/support/pages/backoffice/backoffice-page.ts

```diff
@@ -9,17 +9,22 @@ export class BackofficePage extends AbstractPage {
     cy.visitBackoffice(this.PAGE_URL, options);
   };
 
-  protected interceptTable = (params: InterceptGuiTableParams): void => {
+  public interceptTable = (params: InterceptGuiTableParams, callback?: () => void) => {
     const expectedCount = params.expectedCount ?? 1;
     const interceptAlias = this.faker.string.uuid();
 
     cy.intercept('GET', params.url).as(interceptAlias);
-    cy.wait(`@${interceptAlias}`)
-      .its('response.body.recordsFiltered')
-      .should((total) => {
-        const valueToBeAtMost = expectedCount + Cypress.currentRetry;
-        assert.isTrue(total === expectedCount || total >= valueToBeAtMost);
-      });
+      return cy.wait(`@${interceptAlias}`)
+          .its('response.body.recordsFiltered')
+          .should((total) => {
+              const valueToBeAtMost = expectedCount + Cypress.currentRetry;
```

Comment:
`const valueToBeAtMost = expectedCount + Cypress.currentRetry;`
what is it for? 
example: 
- expectedCount = 3
- Cypress.currentRetry = 2
- valueToBeAtMos = 5

we assert that total (3) >= 5. It will fail, no?

Please change the logic so in general, if the expected count is not provided - we use 0 as default and asser that total >= 0 (default).

Reply:
got the logic, it's done for cases when the test created a new record, and if we rerun the test, BUT doesn't delete previously created - here we need a counter